tests/functional/visibility_unapproved_posts_test.php

```diff
@@ -0,0 +1,360 @@
+		$this->assertContains('Unapproved Posts Test Topic #1', $crawler->filter('h2')->text());
+		$this->assertContains('Re: Unapproved Posts Test Topic #1-#2', $crawler->filter('#page-body')->text());
+		$this->assertContains('This post is not visible to other users until it has been approved', $crawler->filter('#page-body')->text());
+
+		//should be able to see topic 2
+		$crawler = self::request('GET', "viewforum.php?f={$this->data['forums']['Unapproved Posts Test #1']}&sid={$this->sid}");
+		$this->assertContains('Unapproved Posts Test Topic #2', $crawler->filter('html')->text());
+		//should be able to see post in topic 2
+		$crawler = self::request('GET', "viewtopic.php?t={$this->data['topics']['Unapproved Posts Test Topic #2']}&sid={$this->sid}");
+		$this->assertContains('Unapproved Posts Test Topic #2', $crawler->filter('#page-body')->text());
+		$this->assertContains('This post is not visible to other users until it has been approved', $crawler->filter('#page-body')->text());
+		$this->logout();
+
+		// another user
+		$this->login('unapproved_posts_test_user#2');
+
+		$this->add_lang('posting');
+		$this->add_lang('viewtopic');
+		$this->add_lang('mcp');
+
+		//should be able to see topic 1 but not unapproved post
+		$crawler = self::request('GET', "viewtopic.php?t={$this->data['topics']['Unapproved Posts Test Topic #1']}&sid={$this->sid}");
+		$this->assertContains('Unapproved Posts Test Topic #1', $crawler->filter('h2')->text());
+		$this->assertNotContains('Re: Unapproved Posts Test Topic #1-#2', $crawler->filter('#page-body')->text());
+		$this->assertNotContains('This post is not visible to other users until it has been approved', $crawler->filter('#page-body')->text());
+
+		//should not be able to see topic 2
+		$crawler = self::request('GET', "viewforum.php?f={$this->data['forums']['Unapproved Posts Test #1']}&sid={$this->sid}");
+		$this->assertNotContains('Unapproved Posts Test Topic #2', $crawler->filter('html')->text());
+		$this->logout();
+
+		// revert the configuration
+		$this->config_display_unapproved_posts_state(false);
+	}
+
+	public function test_reset_flood_interval()
+	{
+		$this->login();
+		$this->admin_login();
+
+		// Set flood interval back to 15
+		$this->set_flood_interval(15);
+	}
+
+	protected function assert_forum_details($forum_id, $details, $additional_error_message = '')
+	{
+		$this->db = $this->get_db();
+
+		$sql = 'SELECT ' . implode(', ', array_keys($details)) . '
+			FROM phpbb_forums
+			WHERE forum_id = ' . (int) $forum_id;
+		$result = $this->db->sql_query($sql);
+		$data = $this->db->sql_fetchrow($result);
+		$this->db->sql_freeresult($result);
+
+		$this->assertEquals($details, $data, "Forum {$forum_id} does not match expected {$additional_error_message}");
+	}
+
+	protected function set_flood_interval($flood_interval)
+	{
+		$crawler = self::request('GET', 'adm/index.php?sid=' . $this->sid . '&i=acp_board&mode=post');
```

Comment:
You use double quotes in all other places, might aswell here for consistency: 
`"adm/index.php?sid={$this->sid}&i=acp_board&mode=post"`

Reply:
Fixed